CCloud connection authentication progress does not emit websocket connection event updates

When a ccloud connection is created, sidecar is sending an initial websocket CONNECTION_EVENT event with the expected following body payload:

```
2025-01-07 16:18:30.876 [debug] Received websocket connection event 
{
  "action": "CREATED",
  "connection": {
    "api_version": "gateway/v1",
    "kind": "Connection",
    "id": "vscode-confluent-cloud-connection",
    "metadata": {
      "self": "http://localhost:26636/gateway/v1/connections/vscode-confluent-cloud-connection",
      "resource_name": null,
      "sign_in_uri": "https://login.confluent.io/oauth/authorize?...."
    },
    "spec": {
      "id": "vscode-confluent-cloud-connection",
      "name": "Confluent Cloud",
      "type": "CCLOUD"
    },
    "status": {
      "authentication": { "status": "NO_TOKEN" },
      "ccloud": { "state": "NONE" }
    }
  }
}
```

But when the connection's auth status and / or spec.ccloud_config changes when the connection becomes usable, no followup `"action": "UPDATED"` or `"action": "CONNECTED"` websocket event is pushed.

Could someone please consult https://github.com/confluentinc/ide-sidecar/pull/243 and opine as to how to improve so that we can be sidecar push-based as to when a new ccloud connection is fully online, as opposed to polling-based?

I see this in the logs:

```
2025-01-07 16:35:10,505 ERROR [io.con.ide.web.res.ConnectionEvents] (executor-thread-27) Failed to broadcast connection event ConnectionEventBody[action=DISCONNECTED, connection=Connection: java.lang.IllegalArgumentException: Failed to serialize message to JSON
        at io.confluent.idesidecar.websocket.resources.WebsocketEndpoint.broadcast(WebsocketEndpoint.java:220)
        at io.confluent.idesidecar.websocket.resources.WebsocketEndpoint.broadcast(WebsocketEndpoint.java:185)
        at io.confluent.idesidecar.websocket.resources.ConnectionEvents.broadcast(ConnectionEvents.java:55)
        at io.confluent.idesidecar.websocket.resources.ConnectionEvents.broadcast(ConnectionEvents.java:47)
        at io.confluent.idesidecar.websocket.resources.ConnectionEvents.onConnectionDisconnected(ConnectionEvents.java:39)
        at io.confluent.idesidecar.websocket.resources.ConnectionEvents_Observer_onConnectionDisconnected_tHz_Js2PYO3bVSEVJvO9KZnxZYg.notify(Unknown Source)
        at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:351)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:333)
        at io.quarkus.arc.impl.EventImpl$1.get(EventImpl.java:110)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:635)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Java 8 date/time type `java.time.Instant` not supported by default: add Module "com.fasterxml.jackson.datatype:jackson-datatype-jsr310" to enable handling (through reference chain: io.confluent.idesidecar.websocket.messages.Message["body"]->io.confluent.idesidecar.websocket.messages.ConnectionEventBody["connection"]->io.confluent.idesidecar.restapi.models.Connection["status"]->io.confluent.idesidecar.restapi.models.ConnectionStatus["kafka_cluster"]->io.confluent.idesidecar.restapi.models.ConnectionStatus$KafkaClusterStatus["errors"]->io.confluent.idesidecar.restapi.auth.AuthErrors["sign_in"]->io.confluent.idesidecar.restapi.auth.AuthErrors$AuthError["created_at"])
        at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
        at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1330)
        at com.fasterxml.jackson.databind.ser.impl.UnsupportedTypeSerializer.serialize(UnsupportedTypeSerializer.java:35)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:732)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeWithType(BeanSerializerBase.java:653)
        at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:734)
        at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:770)
        at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:183)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:502)
        at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:341)
        at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4799)
        at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:4040)
        at io.confluent.idesidecar.websocket.resources.WebsocketEndpoint.broadcast(WebsocketEndpoint.java:217)
        ... 17 more
```

--- a/src/main/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpoint.java
+++ b/src/main/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpoint.java
@@ -57,6 +57,13 @@ public class WebsocketEndpoint {
   @ConfigProperty(name = "ide-sidecar.websockets.initial-grace-seconds", defaultValue = "30")
   Provider<Integer> initialGraceSeconds;
 
+  /**
+   * Be sure to use the Quarkus-app-wide ObjectMapper instance, configured for extras like
+   * java.time.Instant serialization.
+   */
+  @Inject
+  ObjectMapper mapper;
+
   /** Typesafe wrapper for a websocket session id (string). */
   public static record SessionKey(String sessionId) {
     @Override
@@ -67,19 +74,24 @@ public class WebsocketEndpoint {
 
   /** Wrapper for a websocket session, with additional state tracking. */
   static class WorkspaceWebsocketSession {
+    private final ObjectMapper mapper;
     private final SessionKey key;
     private final Session session;
     private final Instant createdAt;
+
+
 
     /** Will be null if session has not sent HELLO_WORKSPACE message carrying its process id. */
     @Nullable
     private final WorkspacePid workspacePid;
 
-    WorkspaceWebsocketSession(Session session, WorkspacePid workspacePid, Instant createdAt) {
-      key = new SessionKey(session.getId());
+    WorkspaceWebsocketSession(ObjectMapper mapper, Session session, WorkspacePid workspacePid, Instant createdAt) {
+      this.mapper = mapper;
       this.session = session;
+      this.key = new SessionKey(session.getId());
       this.workspacePid = workspacePid;
       this.createdAt = createdAt;
+
     }
 
     /**
@@ -87,7 +99,7 @@ public class WebsocketEndpoint {
      * is 'active'.
      */
     public WorkspaceWebsocketSession buildActive(WorkspacePid workspacePid) {
-      return new WorkspaceWebsocketSession(this.session, workspacePid, this.createdAt);
+      return new WorkspaceWebsocketSession(this.mapper, this.session, workspacePid, this.createdAt);
     }
 
     /** The websocket session id in SessionKey wrapping, used as the key in the sessions map. */
@@ -167,11 +179,6 @@ public class WebsocketEndpoint {
    */
   @Inject
   KnownWorkspacesBean knownWorkspacesBean;
-
-  private final ObjectMapper mapper = JsonMapper
-      .builder()
-      .enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
-      .build();
 
   /**
    * Broadcast a message originating from sidecar to all active connected workspaces.
@@ -291,7 +298,7 @@ public class WebsocketEndpoint {
     // message, at which time we'll replace this mapping with one that carries the pid.
     sessions.put(
         new SessionKey(session.getId()),
-        new WorkspaceWebsocketSession(session, null, Instant.now())
+        new WorkspaceWebsocketSession(this.mapper, session, null, Instant.now())
     );
   }
 

--- a/src/main/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpoint.java
+++ b/src/main/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpoint.java
@@ -79,8 +79,6 @@ public class WebsocketEndpoint {
     private final Session session;
     private final Instant createdAt;
 
-
-
     /** Will be null if session has not sent HELLO_WORKSPACE message carrying its process id. */
     @Nullable
     private final WorkspacePid workspacePid;

--- a/src/test/java/io/confluent/idesidecar/restapi/util/ResourceIOUtil.java
+++ b/src/test/java/io/confluent/idesidecar/restapi/util/ResourceIOUtil.java
@@ -18,7 +18,16 @@ import wiremock.com.github.jknack.handlebars.internal.Files;
 
 public class ResourceIOUtil {
 
-  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static ObjectMapper MAPPER = createObjectMapper();
+
+  private static ObjectMapper createObjectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+    // Automatically registers JavaTimeModule and others as would be done for the ObjectMapper
+    // used within Quarkus will be. Alas that ResourceIOUtil is used as collection
+    // of static methods, @Inject is not possible here.
+    mapper.findAndRegisterModules();
+    return mapper;
+  }
 
   /**
    * Load a resource file at the given path relative to the current thread's context class loader.

--- a/src/test/java/io/confluent/idesidecar/websocket/messages/MessageSerializationTest.java
+++ b/src/test/java/io/confluent/idesidecar/websocket/messages/MessageSerializationTest.java
@@ -54,6 +54,13 @@ public class MessageSerializationTest {
             "websocket-messages/connection-valid.json",
             MessageType.CONNECTION_EVENT,
             ConnectionEventBody.class
+        ),
+        new TestInput(
+            // Has some java.time.Instant fields, not serializable by default
+            // ObjectMapper instances.
+            "websocket-messages/ccloud-connection-connected.json",
+            MessageType.CONNECTION_EVENT,
+            ConnectionEventBody.class
         )
     );
 

--- a/src/test/java/io/confluent/idesidecar/websocket/resources/AbstractWebsocketTestBase.java
+++ b/src/test/java/io/confluent/idesidecar/websocket/resources/AbstractWebsocketTestBase.java
@@ -1,5 +1,6 @@
 package io.confluent.idesidecar.websocket.resources;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import io.confluent.idesidecar.restapi.application.KnownWorkspacesBean;
 import io.confluent.idesidecar.restapi.application.KnownWorkspacesBean.WorkspacePid;
 import io.confluent.idesidecar.restapi.application.SidecarAccessTokenBean;
@@ -23,6 +24,9 @@ abstract class AbstractWebsocketTestBase implements WebsocketClients {
 
   @Inject
   WebsocketEndpoint websocketEndpoint;
+
+  @Inject
+  ObjectMapper mapper;
 
   @TestHTTPResource("/ws")
   URI uri;
@@ -113,6 +117,7 @@ abstract class AbstractWebsocketTestBase implements WebsocketClients {
     TestWebsocketClientConfigurator.setAccessToken(expectedTokenValue);
 
     return connectWorkspace(
+        mapper,
         uri,
         "valid-token",
         workspacePid,

--- a/src/test/java/io/confluent/idesidecar/websocket/resources/WebsocketClients.java
+++ b/src/test/java/io/confluent/idesidecar/websocket/resources/WebsocketClients.java
@@ -56,6 +56,12 @@ public interface WebsocketClients {
     // but when delivered through to other workspaces, the original message type should be preserved.
     private final LinkedBlockingDeque<String> rawMessageStrings = new LinkedBlockingDeque<>();
 
+    private final ObjectMapper mapper;
+
+    public TestWebsocketClientMessageHandler(ObjectMapper mapper) {
+      this.mapper = mapper;
+    }
+
     // For deserializing messages.
     @Override
     public void onMessage(String messageString) {
@@ -64,7 +70,7 @@ public interface WebsocketClients {
       try {
         // Deserialize json -> Message, add to our list of messages so that the main
         // test can check them.
-        Message message = MAPPER.readValue(messageString, Message.class);
+        Message message = mapper.readValue(messageString, Message.class);
         messages.add(message);
       } catch (IOException e) {
         throw new RuntimeException(e);
@@ -132,12 +138,14 @@ public interface WebsocketClients {
 
   /**
    * Finally, a convenience bundle of the above:
-   * 1. a mock workspace process,
-   * 2. its websocket session,
-   * 3. and the message handler that will store messages received by the client.
+   * 1. a reference to the @Injected ObjectMapper,
+   * 2. a mock workspace process,
+   * 3. its websocket session,
+   * 4. The message handler that will store messages received by the client,
    * See {@link #connectWorkspace}
    */
   record ConnectedWorkspace (
+      ObjectMapper mapper,
       WorkspacePid workspacePid,
       Session session,
       TestWebsocketClientMessageHandler messageHandler
@@ -172,7 +180,7 @@ public interface WebsocketClients {
      */
     public ConnectedWorkspace send(Message message)  {
       try {
-        session.getAsyncRemote().sendText(MAPPER.writeValueAsString(message)).get();
+        session.getAsyncRemote().sendText(mapper.writeValueAsString(message)).get();
         Log.info("Test client sent message: " + message);
       } catch (IOException | ExecutionException | InterruptedException e) {
         throw new RuntimeException(e);
@@ -289,8 +297,6 @@ public interface WebsocketClients {
     }
   }
 
-  ObjectMapper MAPPER = new ObjectMapper();
-
   /**
    * Establish a websocket client session, without
    * {@link ConnectedWorkspace#sayHello() binding to a workspace process}.
@@ -303,9 +309,10 @@ public interface WebsocketClients {
   default ConnectedWorkspace connectWorkspace(
       URI uri,
       String authToken,
-      WorkspacePid workspaceProcess
+      WorkspacePid workspaceProcess,
+      ObjectMapper mapper
   ) {
-    return connectWorkspace(uri, authToken, workspaceProcess, false, null);
+    return connectWorkspace(mapper, uri, authToken, workspaceProcess, false, null);
   }
 
   /**
@@ -322,12 +329,15 @@ public interface WebsocketClients {
    * @return A ConnectedWorkspace instance representing the connection.
    */
   default ConnectedWorkspace connectWorkspace(
+      ObjectMapper mapper,
       URI uri,
       String authToken,
       WorkspacePid workspaceProcess,
       boolean sayHello,
       Duration maxTimeToWaitForWorkspaceCount
+
   ) {
+
     if (maxTimeToWaitForWorkspaceCount != null && !sayHello) {
       throw new IllegalArgumentException("maxTimeToWaitForWorkspaceCount only makes sense when saying hello.");
     }
@@ -342,10 +352,10 @@ public interface WebsocketClients {
     }
 
     TestWebsocketClientConfigurator.setAccessToken(authToken);
-    TestWebsocketClientMessageHandler clientHandler = new TestWebsocketClientMessageHandler();
+    TestWebsocketClientMessageHandler clientHandler = new TestWebsocketClientMessageHandler(mapper);
     session.addMessageHandler(clientHandler);
 
-    var workspace = new ConnectedWorkspace(workspaceProcess, session, clientHandler);
+    var workspace = new ConnectedWorkspace(mapper, workspaceProcess, session, clientHandler);
 
     if (sayHello) {
       workspace = workspace.sayHello();

--- a/src/test/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpointTest.java
+++ b/src/test/java/io/confluent/idesidecar/websocket/resources/WebsocketEndpointTest.java
@@ -1,13 +1,17 @@
 package io.confluent.idesidecar.websocket.resources;
 
+import static io.confluent.idesidecar.restapi.util.ResourceIOUtil.loadResourceAsObject;
 import static io.restassured.RestAssured.given;
 import static org.junit.Assert.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import io.confluent.idesidecar.restapi.filters.WorkspaceProcessIdFilter;
 import io.confluent.idesidecar.restapi.testutil.MockWorkspaceProcess;
+import io.confluent.idesidecar.websocket.messages.ConnectionEventBody;
 import io.confluent.idesidecar.websocket.messages.DynamicMessageBody;
 import io.confluent.idesidecar.websocket.messages.HelloBody;
 import io.confluent.idesidecar.websocket.messages.Message;
@@ -18,6 +22,7 @@ import io.confluent.idesidecar.websocket.messages.WorkspacesChangedBody;
 import io.confluent.idesidecar.websocket.resources.WebsocketEndpoint.WorkspaceWebsocketSession;
 import io.quarkus.logging.Log;
 import io.quarkus.test.junit.QuarkusTest;
+import jakarta.inject.Inject;
 import java.io.IOException;
 import java.time.Instant;
 import java.util.List;
@@ -31,6 +36,12 @@ import org.junit.jupiter.params.provider.ValueSource;
  */
 @QuarkusTest
 public class WebsocketEndpointTest extends AbstractWebsocketTestBase {
+
+  // Refer to the same ObjectMapper that quarkus and WebsocketEndpoint uses
+  // when we try to deserialize messages.
+  @Inject
+  ObjectMapper objectMapper;
+
 
   /**
    * Test websocket lifecycle using two mock workspace processes.
@@ -249,6 +260,28 @@ public class WebsocketEndpointTest extends AbstractWebsocketTestBase {
 
     // Then the session should be closed very soon after.
     connectedWorkspace.waitForClose(1000);
+  }
+
+  @Test
+  public void testSendingCCloudConnectionStatusMessage() {
+    // Given a workspace connected happy websocket ...
+    var connectedWorkspace = connectWorkspace(true, true);
+
+    // Load a CCloud connection update with java.time.Instant instances in the body, proving
+    // that websocket-land is using the same well-configured ObjectMapper as the rest of the Quarkus application.
+    var message = loadResourceAsObject("websocket-messages/ccloud-connection-connected.json", Message.class);
+
+    // Send it out from sidecar to all connected workspaces, including the one we just connected.
+    // Should be serialized properly.
+    websocketEndpoint.broadcast(message);
+
+    // wait for the message to be received by the workspace.
+    var recievedMessage = connectedWorkspace.waitForMessageOfType(MessageType.CONNECTION_EVENT, 1000);
+    var receivedBody = (ConnectionEventBody) recievedMessage.body();
+
+    // assert that body.connection.status.authentication.requires_authentication_at round-tripped as a java.time.Instant
+    assertInstanceOf(Instant.class,
+        receivedBody.connection().status().authentication().requiresAuthenticationAt());
   }
 
   /**
@@ -502,6 +535,7 @@ public class WebsocketEndpointTest extends AbstractWebsocketTestBase {
     var maxAllowedSeconds = websocketEndpoint.initialGraceSeconds.get();
 
     var olderSession = new WorkspaceWebsocketSession(
+        this.mapper,
         existingInactiveSession.session(),
         existingInactiveSession.workspacePid(),
         Instant.now().minusSeconds(maxAllowedSeconds + 1) // slightly too old!

--- a/src/test/java/io/confluent/idesidecar/websocket/resources/WorkspaceWebsocketSessionTest.java
+++ b/src/test/java/io/confluent/idesidecar/websocket/resources/WorkspaceWebsocketSessionTest.java
@@ -4,14 +4,21 @@ import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.Mockito.mock;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import io.confluent.idesidecar.restapi.application.KnownWorkspacesBean.WorkspacePid;
 import io.confluent.idesidecar.websocket.resources.WebsocketEndpoint.WorkspaceWebsocketSession;
+import io.quarkus.test.junit.QuarkusTest;
+import jakarta.inject.Inject;
 import jakarta.websocket.Session;
 import java.time.Instant;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+@QuarkusTest
 public class WorkspaceWebsocketSessionTest {
+
+  @Inject
+  ObjectMapper mapper;
 
   /**
    * Test that by default, a WorkspaceWebsocketSession is inactive and has an unknown pid, and that
@@ -22,7 +29,7 @@ public class WorkspaceWebsocketSessionTest {
     // inactive / not yet pid known session
     Instant now = Instant.now();
     var mockSession = getMockSession();
-    WorkspaceWebsocketSession session = new WorkspaceWebsocketSession(mockSession, null, now);
+    WorkspaceWebsocketSession session = new WorkspaceWebsocketSession(this.mapper, mockSession, null, now);
     Assertions.assertFalse(session.isActive());
     // inactive session should report "unknown" for its pid string
     assertEquals("unknown", session.workspacePidString());

--- a/src/test/resources/websocket-messages/ccloud-connection-connected.json
+++ b/src/test/resources/websocket-messages/ccloud-connection-connected.json
@@ -1,0 +1,51 @@
+{
+  "headers": {
+    "message_type": "CONNECTION_EVENT",
+    "originator": "sidecar",
+    "message_id": "4597525f-ac86-4aeb-b656-25658987ff0c"
+  },
+  "body": {
+    "action": "CONNECTED",
+    "connection": {
+      "api_version": "gateway/v1",
+      "kind": "Connection",
+      "id": "vscode-confluent-cloud-connection",
+      "metadata": {
+        "self": "http://localhost:26636/gateway/v1/connections/vscode-confluent-cloud-connection",
+        "resource_name": null,
+        "sign_in_uri": "https://login.confluent.io/oauth/authorize?..."
+      },
+      "spec": {
+        "id": "vscode-confluent-cloud-connection",
+        "name": "Confluent Cloud",
+        "type": "CCLOUD"
+      },
+      "status": {
+        "authentication": {
+          "status": "VALID_TOKEN",
+          "user": {
+            "id": "u-foo",
+            "username": "foo@bar.com",
+            "first_name": "Foo",
+            "last_name": "Bar",
+            "social_connection": "",
+            "auth_type": "AUTH_TYPE_LOCAL"
+          },
+          "requires_authentication_at": "2025-01-09T02:16:28.903418Z"
+        },
+        "ccloud": {
+          "state": "SUCCESS",
+          "user": {
+            "id": "u-foo",
+            "username": "foo@bar.com",
+            "first_name": "Foo",
+            "last_name": "Bar",
+            "social_connection": "",
+            "auth_type": "AUTH_TYPE_LOCAL"
+          },
+          "requires_authentication_at": "2025-01-09T02:16:28.903418Z"
+        }
+      }
+    }
+  }
+}

